SomeSortofCommerce,YouSay/app/src/main/java/com/jonathanlieblich/somesortofcommerceyousay/ProductViewAdapter.java

```diff
@@ -0,0 +1,81 @@
+package com.jonathanlieblich.somesortofcommerceyousay;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.jonathanlieblich.somesortofcommerceyousay.ProductObjects.Product;
+
+import java.util.List;
+
+/**
+ * Created by jonlieblich on 11/1/16.
+ */
+
+public class ProductViewAdapter extends RecyclerView.Adapter<ProductViewHolder> {
+    private List<Product> mProductList;
+
+    public ProductViewAdapter(List<Product> products) {
+        mProductList = products;
+    }
+
+    @Override
+    public ProductViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.product_item, parent, false);
+        return new ProductViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(final ProductViewHolder holder, final int position) {
+        Product product = mProductList.get(position);
+
+        holder.mName.setText(mProductList.get(position).getName());
+        String simplplePrice = mProductList.get(position).getPrice()
+                .substring(1, mProductList.get(position).getPrice().indexOf('D'))+"D";
+        holder.mPrice.setText(simplplePrice);
+
+        int sourceImageS = product.getName().indexOf(' ');
+        int sourceImageC = product.getName().indexOf(',');
+        if(sourceImageC < 0) {
+            sourceImageC = sourceImageS+1;
+        }
+        String sourceImage = sourceImageS > sourceImageC ? product.getName()
+                .substring(0,sourceImageC)+"_main" : product.getName().substring(0, sourceImageS)+"_main";
+        sourceImage = sourceImage.toLowerCase();
+
+        final int thumbnail = holder.mThumbnail.getContext().getResources()
+                .getIdentifier(sourceImage, "drawable", holder.mThumbnail.getContext().getPackageName());
+
+        holder.mThumbnail.setImageResource(thumbnail);
+
+        View.OnClickListener onClick = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                switch(view.getId()) {
+                    case R.id.product_thumbnail:
+                    case R.id.product_name:
+                    case R.id.product_price:
+                    case R.id.single_product_item:
+                        Intent intent = new Intent(view.getContext(), ItemDetailActivity.class);
+                        intent.putExtra("ID", position+1);
```

Comment:
It's better to use `holder.getAdapterPosition()` rather than `position` inside an anonymous inner class like `OnClickListener`. The reason is that this code does not run immediately when `onBindViewHolder()` runs; instead it waits until the user clicks, and by then the position of the item in the `RecyclerView` might have changed.